OLS-425 Sprint-6-Schema

binnyjeshan opened the pull request:
Request you to review the Sprint-6 OLS feature schema

https://telecominfraproject.atlassian.net/wiki/spaces/WIFI/pages/2408087553/Release+4.0+Items

mikehansen1970 reviewed (changes requested):
I would like some clarification based on my comments in the capabilities.
Additionally, please consider the separation of the switch.yml schema into a number of files.

mikehansen1970 commented on capabilities/connect.capabilities.yml:
@Cahb @binnyjeshan 
I don't think this is the right approach.
It seems a great deal of writing has gone into this new capabilities, and I am sure it was with the best intent.
I don't agree with it, but obviously I cannot and will not block it if that's the direction everyone feels it should go.

binnyjeshan commented on schema/switch.yml:
Hi @mikehansen1970, @Cahb  Let us talk about these comments today in the 11/6 Weekly Developer call.

binnyjeshan commented:
Hi @Cahb  @mikehansen1970 

Please review my updated changes based on discussion last week. Summary table of these changes now done are: 
![image](https://github.com/user-attachments/assets/35589f6b-97eb-496f-8617-202cdd151362)

For the capabilities, System limits and abilities is we'd like to convey, and not the real numbers/ranges per configurable or subset of parameters. Such could be handled via config rejections or simply truncations by the OLS-c.
But, capabilities POV, similar to hardware capabilities, it'd be nice if we have the switch advertise what max it can do so that one does not oversubsribe nor tend to undersubscribe in the first place.

mikehansen1970 commented on capabilities/connect.capabilities.yml:
```diff
@@ -139,6 +139,170 @@ properties:
+          - 10G
+          - 40G
+          - 100G
+      dual-active-detection:
```
For things like this, since the configuration is a boolean, would it not make more sense to have an attribute like dual-active-detection be a boolean, as it is enabled, or not, that is all you can do with it configuration wise.

binnyjeshan replied:
I prefer to leave it this way for now. I do see some switches do not support the ICCP yet. In those cases, it'd be good to have it said in the capabilities. I will leave this open for now to discuss in next Sprint. OK with you?

mikehansen1970 commented on capabilities/connect.capabilities.yml:
```diff
@@ -139,6 +139,170 @@ properties:
+          - ICCP
+          - Backup-Link
+          - None
+      failover-time:
```
just call this failover-time-milliseconds?

mikehansen1970 commented on capabilities/connect.capabilities.yml:
```diff
@@ -139,6 +139,170 @@ properties:
+      max-vlans-per-mclag-group:
+        description: Indicates the maximum number of VLANs that can be supported within a single MC-LAG group.
+        type: integer
+      peer-link-bandwidth:
```
I see nowhere to set this in the schema, is it just for reference? Assume so as you would likely have already physically established the links before we get to this point.

mikehansen1970 commented on capabilities/connect.capabilities.yml:
```diff
@@ -139,6 +139,170 @@ properties:
+      lldp-med-support:
+        type: boolean
+        description: Specifies whether LLDP-MED support is available on the switch.
+        examples:
```
Please remove examples

binnyjeshan replied:
I had removed this in my updated code that i pushed last week. Not sure why you didn't see the latest changes.

mikehansen1970 commented on capabilities/connect.capabilities.yml:
```diff
@@ -139,6 +139,170 @@ properties:
+        - 64
+        - 128
+        - 256
+      refresh-interval:
```
maybe rename to refresh-interval-seconds?
As a general comment going forward, not just for this change, if there is a time related value, specify the unit (seconds, milliseconds, minutes, etc) in the name would immediately add clarity

mikehansen1970 commented on capabilities/connect.capabilities.yml:
```diff
@@ -139,6 +139,170 @@ properties:
+            examples:
+            - 10
+            - 20
+      reinit-delay:
```
again, add time value to name?

binnyjeshan replied:
I had removed this in my updated code that i pushed last week. Not sure why you didn't see the latest changes.

mikehansen1970 commented on capabilities/connect.capabilities.yml:
```diff
@@ -139,6 +139,170 @@ properties:
+            examples:
+            - 5
+            - 10
+      notification-interval:
```
again time value if we are going to use this

binnyjeshan replied:
This was removed last week when i raised second review. I'll push changes now today, for you to recheck final changes.

mikehansen1970 commented on schema/ethernet.yml:
```diff
@@ -168,6 +168,14 @@ properties:
+      mac-address-bypass:
+        description: Trigger mac-auth when a new ARP is learned.
+        type: boolean
+      mac-address-bypass-timout:
```
What is the unit, seconds? milliseconds?

Cahb replied:
Also, i think there's a typo?
>      mac-address-bypass-timout
..
>      mac-address-bypass-tim**e**out

binnyjeshan replied:
mac-address-bypass:
        description: Enables bypass when a device does not support 802.1X authentication (e.g., printers, IP phones)
        type: boolean
      **mac-address-bypass-timeout:**
        description: Defines the time period **(in minutes)** for which a MAC address is allowed access to the network without requiring reauthentication, after being authenticated or allowed via MAC Authentication Bypass (MAB).
        type: integer

mikehansen1970 reviewed (commented):
I have added some additional comments

Cahb commented on schema/ethernet.yml:
```diff
@@ -168,6 +168,14 @@ properties:
+        description: Trigger mac-auth when a new ARP is learned.
+        type: boolean
+      mac-address-bypass-timout:
+        description: Trigger mac-auth when a new ARP is learned.
```
(minor comment)
Also, shouldn't this description be a bit different from the field above?
Since it's a timeout value, not an actual option to <trigger mac-auth...>

binnyjeshan replied:
This was a typo. Will correct.

Cahb commented on schema/ethernet.yml:
```diff
@@ -234,3 +242,129 @@ properties:
+          - receive
+          - transmit
+          - tx-rx
+          - disabled
```
@mikehansen1970 @binnyjeshan
Just a comment for discussion: do you think maybe we have to skip / stop adding <disabled> stuff into schema and stick to
<if it's not present in config request, it should be treated as a default - disabled / or whatever default is>?

So like here if we'd push a config request with lldp-admin-status omitted, it should be disabled
Or it's a too <shallow> approach, if you may, or not _that_ obvious and transparent?

mikehansen1970 replied:
@Cahb @binnyjeshan 
This is a workable approach as long as it is clear. It might be better to have it set to whatever the default is, I am not sure. For some things, disabled, or just not configured, makes sense.
At a higher level though, for example, if you send only 1 port in the ethernet block, would that implicitly disable all the other ports? That needs clarity.

Cahb replied:
(small note)
Also, on the other note:
```
- receive
- transmit
- tx-rx
```
Should we maybe make them aligned:
```
- rx
- tx
- rx-tx
```

binnyjeshan replied:
Will incorporate this.

binnyjeshan commented on schema/switch.yml:
```diff
@@ -509,4 +551,166 @@ properties:
+    properties:
+      voice-vlan-enable:
+        description: "This option specifically enables or disables the Voice VLAN feature for the system. When enabled, the switch starts recognizing and assigning voice traffic to the designated Voice VLAN."
+        type: boolean
```
Okay, will remove this.

binnyjeshan commented:
@Cahb @mikehansen1970 

I have addressed review comments like this.  Please check and let me know how it looks. 

![image](https://github.com/user-attachments/assets/f3f0e63d-f330-4a7c-a301-37854c6eb048)
